### PR TITLE
AccessControl: Protect /datasources endpoints consistently with NavLinks permissions

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:
There is an inconsistency in the `/datasources` front end endpoints access permissions. This PR intends to fix this.
Basically to be consistent with the rights I need to have to access the `/datasources/` `/datasources/edit` `/datasources/new` pages using the NavBar, I need to use the following evaluators:
```go
// dataSourcesConfigurationAccessEvaluator is used to protect the "Configure > Data sources" tab access
var dataSourcesConfigurationAccessEvaluator = ac.EvalAll(
	ac.EvalPermission(ActionDatasourcesRead, ScopeDatasourcesAll),
	ac.EvalAny(
		ac.EvalPermission(ActionDatasourcesCreate),
		ac.EvalPermission(ActionDatasourcesDelete),
		ac.EvalPermission(ActionDatasourcesWrite),
	),
)

// dataSourcesNewAccessEvaluator is used to protect the "Configure > Data sources > New" page access
var dataSourcesNewAccessEvaluator = ac.EvalAll(
	ac.EvalPermission(ActionDatasourcesRead, ScopeDatasourcesAll),
	ac.EvalPermission(ActionDatasourcesCreate),
	ac.EvalPermission(ActionDatasourcesWrite),
)

// dataSourcesEditAccessEvaluator is used to protect the "Configure > Data sources > Edit" page access
var dataSourcesEditAccessEvaluator = ac.EvalAll(
	ac.EvalPermission(ActionDatasourcesRead, ScopeDatasourcesAll),
	ac.EvalPermission(ActionDatasourcesWrite),
)
```

More info can be found in https://github.com/grafana/grafana-enterprise/issues/1916

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1916

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/index.go [hunk 1]
```diff
@@ -28,6 +28,19 @@ var dataSourcesConfigurationAccessEvaluator = ac.EvalAll(
 	),
 )
 
+// dataSourcesNewAccessEvaluator is used to protect the "Configure > Data sources > New" page access
```
Comment (gamab) [inline pkg/api/index.go:31]: @alexanderzobnin, @kalleep, @sakjur:
Not sure where I should actually put these evaluators. In the `roles.go` file, maybe? In their own `evaluator.go` file?
Wdyt?

**Review by vtorosyan — APPROVED** [review]
Makes sense 👍

File: pkg/api/roles.go [hunk 2]
```diff
@@ -90,3 +90,29 @@ func (hs *HTTPServer) declareFixedRoles() error {
...
+		accesscontrol.EvalPermission(ActionDatasourcesDelete),
+		accesscontrol.EvalPermission(ActionDatasourcesWrite),
+	),
+)
+
+// dataSourcesNewAccessEvaluator is used to protect the "Configure > Data sources > New" page access
+var dataSourcesNewAccessEvaluator = accesscontrol.EvalAll(
+	accesscontrol.EvalPermission(ActionDatasourcesRead, ScopeDatasourcesAll),
+	accesscontrol.EvalPermission(ActionDatasourcesCreate),
+	accesscontrol.EvalPermission(ActionDatasourcesWrite),
```
Comment (sakjur) [inline pkg/api/roles.go:111]: Do you need write?

**Review by sakjur — APPROVED** [review]

File: pkg/api/roles.go (hunk 2 above)
Comment (gamab) [reply pkg/api/roles.go:111]: Yes. If you get access to this page you would be able to create a data source but not edit its values, resulting in the data source having default values.
https://github.com/grafana/grafana/discussions/38924

**Review by IevaVasiljeva — APPROVED** [review]
Looks good as far as I can tell :)

File: pkg/api/index.go (hunk 1 above)
Comment (IevaVasiljeva) [reply pkg/api/index.go]: Do you use them anywhere else? I currently have a set of licensing read permissions that are being evaluated in both `licensing/api/api.go` and `api/frontendsettings.go`, so I could benefit from having an evaluator for them somewhere in OSS, but I'm not sure whether there's a good place for it.

**Pull request merged**